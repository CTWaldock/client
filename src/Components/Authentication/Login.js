import React, { Fragment } from 'react'
import Form from '../../UI/Form'
import FormField from '../../UI/FormField'
import {StyledLink, ErrorMessage, Center, Button} from './Auth.styles.js'



const Login = (props) => {
    // console.log(props)
    const { loginError, login, url } = props
    const {user} = localStorage

    const localName = user ? `Hello, ${user.firstName}` : 'Hello Stranger!'

    return (
        <Fragment>
            <Center>
                <Form onSubmit={(e) => login(e, url)} title={localName}>
                    <FormField label="Email" name="email" type="email" onChange="" />
                    <FormField label="Password" name="password" type="password" onChange="" />
                    <ErrorMessage>{loginError}</ErrorMessage>
                    <Button className="formButton">Sign In</Button>
                    <StyledLink to="/register">Need to register a new account?</StyledLink>
                </Form>
            </Center>
        </Fragment>
    )
}

export default Login