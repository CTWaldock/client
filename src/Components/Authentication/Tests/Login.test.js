import React from 'react'
import Login from '../Login'
import renderer from 'react-test-renderer'
import {BrowserRouter as Router} from 'react-router-dom'
import localStorage from '../../../localStorage'

window.localStorage = localStorage

describe('Login', () => {
    it('Should render the way we expect', () => {
        
        
        const tree = renderer.create(
            <Router>
                <Login registerError='something' register='something' url='something' />
            </Router>
        )
        expect(tree.toJSON()).toMatchSnapshot()
    })
})
