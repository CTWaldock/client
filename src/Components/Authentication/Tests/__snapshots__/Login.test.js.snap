// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Should render the way we expect 1`] = `
<div
  className="sc-VigVT gUteem"
>
  <form
    className="sc-gzVnrw eqCBFZ"
    id={undefined}
    onSubmit={[Function]}
  >
    <h3
      className="sc-htoDjs fbZTTv"
    >
      Hello Stranger!
    </h3>
    <label
      className="sc-dnqmqq gxITcO"
    >
      Email
      <span />
    </label>
    <input
      autoComplete="nope"
      autoFocus={null}
      className="sc-iwsKbI kxDmNQ"
      name="email"
      onChange={null}
      placeholder={undefined}
      step={null}
      type="email"
    />
    <label
      className="sc-dnqmqq gxITcO"
    >
      Password
      <span />
    </label>
    <input
      autoComplete="nope"
      autoFocus={null}
      className="sc-iwsKbI kxDmNQ"
      name="password"
      onChange={null}
      placeholder={undefined}
      step={null}
      type="password"
    />
    <p
      className="sc-gqjmRU cKmdHu"
    />
    <button
      className="formButton sc-jTzLTM ffePeW"
    />
    <a
      className="sc-gZMcBi htfulI"
      href="/register"
      onClick={[Function]}
    >
      Need to register a new account?
    </a>
    <div />
  </form>
</div>
`;
